samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -153,23 +170,67 @@ class StreamConfig(config: Config) extends ScalaMapConfig(config) with Logging {
   }
 
   /**
-    * Gets the specified property for a SystemStream.
+    * Gets the specified Samza property for a SystemStream. A Samza property is a property that controls how Samza
+    * interacts with the stream, as opposed to a property of the stream itself.
     *
     * Note, because the translation is not perfect between SystemStream and streamId,
     * this method is not identical to getProperty(streamId, property)
+    *
+    * @param systemStream the SystemStream for which the property value will be retrieved.
+    * @param property the samza property name excluding the leading delimiter. e.g. "samza.x.y"
     */
-  private def getProperty(systemStream: SystemStream, property: String): String = {
-    val streamVal = getStreamProperties(systemStreamToStreamId(systemStream)).get(property)
+  def getSamzaProperty(systemStream: SystemStream, property: String): String = {
```

Comment:
shall we make this private/package private so we will only expose the getters for specific configs?